awx/main/middleware.py

```diff
@@ -136,6 +137,12 @@ def process_response(self, request, response):
             expiry = int(settings.SESSION_COOKIE_AGE)
             request.session.set_expiry(expiry)
             response['Session-Timeout'] = expiry
+
+        token = getattr(request.user, 'oauth_token', None)
```

Comment:
Middleware _feels_ like the right place to do this, but I'm open to other suggestions.

Reply:
4d41e316392bb49f1164c969532b6632a5589d49

Reply:
I think middleware is better than in a commit hook which can be obscure and hard to find. The other place may be in the `APIView` generic, but I think middleware is the right place for this.

Reply:
Eh, I actually agree with @AlanCoding, I think `connection.on_commit` is a lot cleaner.

Reply:
https://docs.djangoproject.com/en/2.2/topics/db/transactions/

> Middleware runs outside of the transaction

Running this outside of the request transaction is the only really important thing here. It looks like either of the solutions will satisfy this.